support/controlplane-component/status.go

```diff
@@ -21,14 +21,20 @@ import (
 )
 
 const (
-	KubeAPIServerComponentName = "kube-apiserver"
+	kubeAPIServerComponentName = "kube-apiserver"
+	etcdComponentName          = "etcd"
 )
 
 func (c *controlPlaneWorkload) checkDependencies(cpContext ControlPlaneContext) ([]string, error) {
 	unavailableDependencies := sets.New(c.dependencies...)
-	// always add kube-apiserver as a dependency.
-	// TODO: don't add for etcd component once refactored.
-	unavailableDependencies.Insert(KubeAPIServerComponentName)
+	// always add kube-apiserver as a dependency, except for etcd.
```

Comment:
can we please unit test this func?

Reply:
created a ticket https://issues.redhat.com/browse/HOSTEDCP-2068